src/components/TableOfContents/index.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from 'react';
+import { _cs } from '@togglecorp/fujs';
+
+import PrimaryButton from '#rsca/Button/PrimaryButton';
+import ListView from '#rsu/../v2/View/ListView';
+
+import styles from './styles.css';
+
+interface Props<T>{
+    keySelector: (datum: T) => string | number;
+    labelSelector: (datum: T) => string | number;
+    childrenSelector: (datum: T) => T[] | undefined;
+    onChange: (key: string | number) => void;
+    options: T[];
+    value: string | number | undefined;
+}
```

Comment:
I think we should have a generic type for key.

```suggestion
interface Props<T, K extends string | number>{
    keySelector: (datum: T) => string | number;
    labelSelector: (datum: T) => K;
    childrenSelector: (datum: T) => T[] | undefined;
    onChange: (key: K) => void;
    options: T[];
    value: string | number | undefined;
}
```